File: app/src/main/java/net/tlalka/android/fiszki/injections/modules/ApplicationModule.java
package net.tlalka.android.fiszki.injections.modules;

import android.app.Application;
import dagger.Module;
import dagger.Provides;
import net.tlalka.android.fiszki.navigations.Navigator;

import javax.inject.Singleton;

@Module
public class ApplicationModule {

    private final Application application;

    public ApplicationModule(Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    Application getApplication() {
        return this.application;
    }

    @Provides
    @Singleton
    Navigator getNavigator() {
        return new Navigator();
    }
}

File: app/src/main/kotlin/net/tlalka/fiszki/core/modules/ApplicationModule.kt
package net.tlalka.fiszki.core.modules

import android.app.Application
import dagger.Module
import dagger.Provides
import net.tlalka.fiszki.core.annotations.ApplicationScope
import net.tlalka.fiszki.view.navigations.Navigator

@Module
class ApplicationModule(@get:Provides @get:ApplicationScope val application: Application) {

    @Provides
    @ApplicationScope
    fun getNavigator(): Navigator {
        return Navigator()
    }
}
